nrf_security/src/mbedcrypto_glue/cc310/CMakeLists.txt

```diff
@@ -4,24 +4,29 @@
 # SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
 #
 if(CONFIG_CC310_BACKEND)
+
   zephyr_library_named(mbedcrypto_glue_cc310)
-  zephyr_library_sources(
-      ${CMAKE_CURRENT_LIST_DIR}/aes_cc310.c
-      ${CMAKE_CURRENT_LIST_DIR}/ccm_cc310.c
-      ${CMAKE_CURRENT_LIST_DIR}/cmac_cc310.c
-      ${CMAKE_CURRENT_LIST_DIR}/dhm_cc310.c
-      ${CMAKE_CURRENT_LIST_DIR}/ecdh_cc310.c
-      ${CMAKE_CURRENT_LIST_DIR}/ecdsa_cc310.c
-      ${CMAKE_CURRENT_LIST_DIR}/rsa_cc310.c
-  )
+
+  #
+  # CC310: Add glued files if they are enabled
+  #
+  zephyr_library_sources_ifdef(CONFIG_CC310_MBEDTLS_AES_C   ${CMAKE_CURRENT_LIST_DIR}/aes_cc310.c)
+  zephyr_library_sources_ifdef(CONFIG_CC310_MBEDTLS_CCM_C   ${CMAKE_CURRENT_LIST_DIR}/ccm_cc310.c)
+  zephyr_library_sources_ifdef(CONFIG_CC310_MBEDTLS_DHM_C   ${CMAKE_CURRENT_LIST_DIR}/dhm_cc310.c)
```

Comment:
I'm observing something strange. These sources are not being compiled even though they are enabled.

To reproduce build the hello_world sample with board nrf52840_pca10056 and this prj.conf file:

`# Generated by Kconfiglib (https://github.com/ulfalizer/Kconfiglib)

#
# Modules
#

#
# Nordic nRF Connect
#
# CONFIG_MCUBOOT_BUILD_S1_VARIANT is not set
# CONFIG_SECURE_BOOT_CRYPTO_CLIENT is not set
# CONFIG_SECURE_BOOT is not set
# CONFIG_IS_SECURE_BOOTLOADER is not set

#
# Bluetooth Low Energy
#
# end of Bluetooth Low Energy

# CONFIG_NRF_ESB is not set

#
# Networking
#

#
# Application protocols
#
# CONFIG_MQTT_SOCKET_LIB is not set
# CONFIG_NRF_CLOUD is not set
# CONFIG_DOWNLOAD_CLIENT is not set
# CONFIG_FOTA_DOWNLOAD is not set
# CONFIG_AWS_JOBS is not set
# CONFIG_AWS_FOTA is not set
# CONFIG_CLOUD_API is not set
# end of Application protocols
# end of Networking

#
# NFC
#
# CONFIG_NFC_NDEF is not set
# CONFIG_NFC_NDEF_PARSER is not set
# CONFIG_NFC_T2T_PARSER is not set
# CONFIG_NFC_T4T_ISODEP is not set
# end of NFC

# CONFIG_PROFILER is not set

#
# SPM
#
# CONFIG_SPM is not set
# CONFIG_IS_SPM is not set
# end of SPM

# CONFIG_FW_METADATA is not set
# CONFIG_PPI_TRACE is not set

#
# Libraries
#

#
# BSD Library for nrf91
#
# end of BSD Library for nrf91

#
# AT Host Library for nrf91
#
# CONFIG_AT_HOST_LIBRARY is not set
# end of AT Host Library for nrf91

# CONFIG_DK_LIBRARY is not set
# CONFIG_AT_CMD_PARSER is not set
# CONFIG_MODEM_INFO is not set
# end of Libraries

#
# Device Drivers
#
# CONFIG_ADP536X is not set

#
# AT Command driver
#
# end of AT Command driver

# CONFIG_GPS_SIM is not set
# CONFIG_LTE_LINK_CONTROL is not set
# CONFIG_ETH_RTT is not set
# CONFIG_SENSOR is not set
# CONFIG_ST25R3911B_LIB is not set
# CONFIG_CLOCK_CONTROL_NRF_K32SRC_RC is not set
CONFIG_CLOCK_CONTROL_NRF_K32SRC_XTAL=y
# CONFIG_CLOCK_CONTROL_NRF_K32SRC_SYNTH is not set
# CONFIG_CLOCK_CONTROL_NRF_K32SRC_BLOCKING is not set
# CONFIG_CLOCK_CONTROL_NRF_K32SRC_500PPM is not set
# CONFIG_CLOCK_CONTROL_NRF_K32SRC_250PPM is not set
# CONFIG_CLOCK_CONTROL_NRF_K32SRC_150PPM is not set
# CONFIG_CLOCK_CONTROL_NRF_K32SRC_100PPM is not set
# CONFIG_CLOCK_CONTROL_NRF_K32SRC_75PPM is not set
# CONFIG_CLOCK_CONTROL_NRF_K32SRC_50PPM is not set
# CONFIG_CLOCK_CONTROL_NRF_K32SRC_30PPM is not set
CONFIG_CLOCK_CONTROL_NRF_K32SRC_20PPM=y
# CONFIG_CLOCK_CONTROL_NRF_K32SRC_10PPM is not set
# CONFIG_CLOCK_CONTROL_NRF_K32SRC_5PPM is not set
# CONFIG_CLOCK_CONTROL_NRF_K32SRC_2PPM is not set
# CONFIG_CLOCK_CONTROL_NRF_K32SRC_1PPM is not set
# end of Device Drivers

#
# External libraries
#
# CONFIG_CJSON_LIB is not set
# end of External libraries

# CONFIG_UNITY is not set
# end of Nordic nRF Connect

#
# MCUboot
#
CONFIG_BOOT_SIGNATURE_KEY_FILE="root-rsa-2048.pem"
# end of MCUboot

#
# Nordic nrfxlib
#
# CONFIG_NRFXLIB_NFC is not set

#
# Crypto libraries for nRF5x SOCs.
#
CONFIG_NRFXLIB_CRYPTO=y
# CONFIG_SHORT_WCHAR_T is not set
# CONFIG_NRF_OBERON is not set
# CONFIG_NRF_CC310_BL is not set
CONFIG_NRF_CC310_PLATFORM=y
# end of Crypto libraries for nRF5x SOCs.

#
# Nordic Security
#
CONFIG_NORDIC_SECURITY_BACKEND=y

#
# mbed TLS Configuration File
#
CONFIG_MBEDTLS_CFG_FILE="nrf-config.h"
CONFIG_GENERATE_MBEDTLS_CFG_FILE=y
CONFIG_MBED_X509_LIBRARY=y
CONFIG_MBED_TLS_LIBRARY=y

#
# Backend Selection
#
CONFIG_CC310_BACKEND=y
CONFIG_MBEDTLS_VANILLA_BACKEND=y
CONFIG_NRF_CRYPTO_BACKEND_COMBINATION_0=y

#
# Configuration section
#

#
# AES   - Advanced Encryption Standard
#

#
# AES block cipher common, including ECB - Electronic Code Book mode
#
CONFIG_CC310_MBEDTLS_AES_C=y
# CONFIG_VANILLA_MBEDTLS_AES_C is not set
# end of AES block cipher common, including ECB - Electronic Code Book mode

CONFIG_MBEDTLS_CIPHER_MODE_CBC=y

#
# CBC cipher padding modes
#
CONFIG_MBEDTLS_CIPHER_PADDING_PKCS7=y
CONFIG_MBEDTLS_CIPHER_PADDING_ONE_AND_ZEROS=y
CONFIG_MBEDTLS_CIPHER_PADDING_ZEROS_AND_LEN=y
CONFIG_MBEDTLS_CIPHER_PADDING_ZEROS=y
# end of CBC cipher padding modes

CONFIG_MBEDTLS_CIPHER_MODE_CTR=y
CONFIG_MBEDTLS_CIPHER_MODE_CFB=y
CONFIG_MBEDTLS_CIPHER_MODE_OFB=y
CONFIG_MBEDTLS_CMAC_C=y
CONFIG_CC310_MBEDTLS_CMAC_C=y
# CONFIG_VANILLA_MBEDTLS_CMAC_C is not set
# end of AES   - Advanced Encryption Standard

#
# AEAD  - Authenticated Encryption with Associated Data
#

#
# AES-CCM - Counter with CBC-MAC mode
#
CONFIG_CC310_MBEDTLS_CCM_C=y
# end of AES-CCM - Counter with CBC-MAC mode

CONFIG_MBEDTLS_CHACHA20_C=y
CONFIG_CC310_MBEDTLS_CHACHA20_C=y
# CONFIG_VANILLA_MBEDTLS_CHACHA20_C is not set
CONFIG_MBEDTLS_POLY1305_C=y
CONFIG_CC310_MBEDTLS_POLY1305_C=y
# CONFIG_VANILLA_MBEDTLS_POLY1305_C is not set
CONFIG_MBEDTLS_CHACHAPOLY_C=y
CONFIG_CC310_MBEDTLS_CHACHAPOLY_C=y
# CONFIG_VANILLA_MBEDTLS_CHACHAPOLY_C is not set
# end of AEAD  - Authenticated Encryption with Associated Data

#
# DHM   - Diffie-Hellman-Merkel
#
CONFIG_CC310_MBEDTLS_DHM_C=y
# CONFIG_VANILLA_MBEDTLS_DHM_C is not set
# end of DHM   - Diffie-Hellman-Merkel

#
# ECC - Elliptic Curve Cryptography
#
CONFIG_MBEDTLS_ECP_C=y
CONFIG_CC310_MBEDTLS_ECP_C=y
# CONFIG_VANILLA_MBEDTLS_ECP_C is not set
CONFIG_MBEDTLS_ECDH_C=y
CONFIG_CC310_MBEDTLS_ECDH_C=y
# CONFIG_VANILLAMBEDTLS_ECDH_C is not set
CONFIG_MBEDTLS_ECDSA_C=y
CONFIG_CC310_MBEDTLS_ECDSA_C=y
# CONFIG_VANILLA_MBEDTLS_ECDSA_C is not set
CONFIG_CC310_MBEDTLS_ECJPAKE_C=y
CONFIG_MBEDTLS_ECJPAKE_C=y

#
# ECC curves
#
# CONFIG_MBEDTLS_ECP_DP_SECP192R1_ENABLED is not set
# CONFIG_MBEDTLS_ECP_DP_SECP224R1_ENABLED is not set
CONFIG_MBEDTLS_ECP_DP_SECP256R1_ENABLED=y
# CONFIG_MBEDTLS_ECP_DP_SECP384R1_ENABLED is not set
# CONFIG_MBEDTLS_ECP_DP_SECP521R1_ENABLED is not set
# CONFIG_MBEDTLS_ECP_DP_SECP192K1_ENABLED is not set
# CONFIG_MBEDTLS_ECP_DP_SECP224K1_ENABLED is not set
# CONFIG_MBEDTLS_ECP_DP_SECP256K1_ENABLED is not set
# CONFIG_MBEDTLS_ECP_DP_CURVE25519_ENABLED is not set
# end of ECC curves
# end of ECC - Elliptic Curve Cryptography

#
# RSA   - Rivest–Shamir–Adleman cryptosystem
#
CONFIG_MBEDTLS_RSA_C=y
CONFIG_CC310_MBEDTLS_RSA_C=y
# CONFIG_VANILLA_MBEDTLS_RSA_C is not set
# CONFIG_MBEDTLS_PKCS1_V15 is not set
CONFIG_MBEDTLS_PKCS1_V21=y
# end of RSA   - Rivest–Shamir–Adleman cryptosystem

#
# SHA   - Secure Hash Algorithm
#
CONFIG_MBEDTLS_SHA1_C=y
CONFIG_CC310_MBEDTLS_SHA1_C=y
# CONFIG_VANILLA_MBEDTLS_SHA1_C is not set
CONFIG_MBEDTLS_SHA256_C=y
CONFIG_CC310_MBEDTLS_SHA256_C=y
# CONFIG_VANILLA_MBEDTLS_SHA256_C is not set
CONFIG_MBEDTLS_SHA512_C=y
# end of SHA   - Secure Hash Algorithm

CONFIG_APP_LINK_WITH_MBEDCRYPTO_INCLUDES=y
# end of Nordic Security
# end of Nordic nrfxlib

#
# Optional modules. Make sure they're installed, via the project manifest.
#
# CONFIG_CIVETWEB is not set
# CONFIG_LIBMETAL is not set
# CONFIG_MBEDTLS is not set
# CONFIG_MBEDTLS_TLS_VERSION_1_0 is not set
# CONFIG_MBEDTLS_TLS_VERSION_1_1 is not set
CONFIG_MBEDTLS_TLS_VERSION_1_2=y
# CONFIG_MBEDTLS_DTLS is not set
# CONFIG_MBEDTLS_SSL_EXPORT_KEYS is not set
# CONFIG_MBEDTLS_KEY_EXCHANGE_ALL_ENABLED is not set
# CONFIG_MBEDTLS_KEY_EXCHANGE_PSK_ENABLED is not set
# CONFIG_MBEDTLS_KEY_EXCHANGE_DHE_PSK_ENABLED is not set
# CONFIG_MBEDTLS_KEY_EXCHANGE_ECDHE_PSK_ENABLED is not set
# CONFIG_MBEDTLS_KEY_EXCHANGE_RSA_PSK_ENABLED is not set
CONFIG_MBEDTLS_KEY_EXCHANGE_RSA_ENABLED=y
# CONFIG_MBEDTLS_KEY_EXCHANGE_DHE_RSA_ENABLED is not set
# CONFIG_MBEDTLS_KEY_EXCHANGE_ECDHE_RSA_ENABLED is not set
# CONFIG_MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED is not set
# CONFIG_MBEDTLS_KEY_EXCHANGE_ECDH_ECDSA_ENABLED is not set
# CONFIG_MBEDTLS_KEY_EXCHANGE_ECDH_RSA_ENABLED is not set
# CONFIG_MBEDTLS_KEY_EXCHANGE_ECJPAKE_ENABLED is not set
# CONFIG_MBEDTLS_CIPHER_ALL_ENABLED is not set
CONFIG_MBEDTLS_CIPHER_AES_ENABLED=y
CONFIG_MBEDTLS_AES_ROM_TABLES=y
# CONFIG_MBEDTLS_CIPHER_CAMELLIA_ENABLED is not set
CONFIG_MBEDTLS_CIPHER_DES_ENABLED=y
# CONFIG_MBEDTLS_CIPHER_ARC4_ENABLED is not set
# CONFIG_MBEDTLS_CIPHER_CHACHA20_ENABLED is not set
# CONFIG_MBEDTLS_CIPHER_BLOWFISH_ENABLED is not set
# CONFIG_MBEDTLS_CIPHER_CCM_ENABLED is not set
# CONFIG_MBEDTLS_CIPHER_MODE_XTS_ENABLED is not set
# CONFIG_MBEDTLS_CIPHER_MODE_GCM_ENABLED is not set
CONFIG_MBEDTLS_CIPHER_CBC_ENABLED=y
# CONFIG_MBEDTLS_MAC_ALL_ENABLED is not set
# CONFIG_MBEDTLS_MAC_MD4_ENABLED is not set
CONFIG_MBEDTLS_MAC_MD5_ENABLED=y
CONFIG_MBEDTLS_MAC_SHA1_ENABLED=y
CONFIG_MBEDTLS_MAC_SHA256_ENABLED=y
CONFIG_MBEDTLS_SHA256_SMALLER=y
# CONFIG_MBEDTLS_MAC_SHA512_ENABLED is not set
# CONFIG_MBEDTLS_MAC_POLY1305_ENABLED is not set
# CONFIG_MBEDTLS_MAC_CMAC_ENABLED is not set
CONFIG_MBEDTLS_CTR_DRBG_ENABLED=y
# CONFIG_MBEDTLS_HMAC_DRBG_ENABLED is not set
# CONFIG_MBEDTLS_GENPRIME_ENABLED is not set
# CONFIG_MBEDTLS_PEM_CERTIFICATE_FORMAT is not set
# CONFIG_MBEDTLS_HAVE_ASM is not set
# CONFIG_MBEDTLS_ENTROPY_ENABLED is not set
# CONFIG_MBEDTLS_USER_CONFIG_ENABLE is not set
# CONFIG_OPENAMP is not set

#
# QMSI
#
# end of QMSI

# CONFIG_TINYCBOR is not set

#
# TLS configuration
#

#
# Supported TLS version
#
# end of Supported TLS version

#
# Ciphersuite configuration
#

#
# Supported key exchange modes
#

#
# Supported cipher modes
#

#
# Supported message authentication methods
#
# end of Ciphersuite configuration

#
# Random number generators
#

#
# Other configurations
#
# end of TLS configuration
# end of Modules

CONFIG_BOARD="nrf52840_pca10056"
CONFIG_SOC="nRF52840_QIAA"
CONFIG_SOC_SERIES="nrf52"
CONFIG_NUM_IRQS=48
CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC=32768
# CONFIG_WATCHDOG is not set
# CONFIG_UART_NS16550 is not set
CONFIG_GPIO=y
CONFIG_ISR_STACK_SIZE=2048
CONFIG_NET_CONFIG_IEEE802154_DEV_NAME=""
CONFIG_CLOCK_CONTROL=y
CONFIG_NRF_RTC_TIMER=y
CONFIG_SYS_CLOCK_TICKS_PER_SEC=32768
CONFIG_ARCH_HAS_CUSTOM_BUSY_WAIT=y
CONFIG_SYS_POWER_MANAGEMENT=y
CONFIG_BUILD_OUTPUT_HEX=y
CONFIG_TEXT_SECTION_OFFSET=0
CONFIG_FLASH_SIZE=1024
CONFIG_FLASH_BASE_ADDRESS=0x0
CONFIG_SRAM_SIZE=256
CONFIG_SRAM_BASE_ADDRESS=0x20000000
# CONFIG_SPI is not set
CONFIG_SOC_GECKO_EMU=y
CONFIG_BOARD_NRF52840_PCA10056=y

#
# Board Options
#
CONFIG_BOARD_ENABLE_DCDC=y
# end of Board Options

# CONFIG_SOC_SERIES_BEETLE is not set
# CONFIG_SOC_SERIES_MPS2 is not set
# CONFIG_SOC_SERIES_MUSCA is not set
# CONFIG_SOC_SERIES_MUSCA_B1 is not set
# CONFIG_SOC_SERIES_SAM3X is not set
# CONFIG_SOC_SERIES_SAM4S is not set
# CONFIG_SOC_SERIES_SAME70 is not set
# CONFIG_SOC_SERIES_SAMD20 is not set
# CONFIG_SOC_SERIES_SAMD21 is not set
# CONFIG_SOC_SERIES_SAMR21 is not set
# CONFIG_SOC_SERIES_PSOC62 is not set
# CONFIG_SOC_SERIES_MEC1501X is not set
# CONFIG_SOC_SERIES_MEC1701X is not set
# CONFIG_SOC_SERIES_NRF51X is not set
CONFIG_SOC_SERIES_NRF52X=y
# CONFIG_SOC_SERIES_NRF91X is not set
# CONFIG_SOC_SERIES_IMX_6X_M4 is not set
# CONFIG_SOC_SERIES_IMX7_M4 is not set
# CONFIG_SOC_SERIES_IMX_RT is not set
# CONFIG_SOC_SERIES_KINETIS_K6X is not set
# CONFIG_SOC_SERIES_KINETIS_KE1XF is not set
# CONFIG_SOC_SERIES_KINETIS_KL2X is not set
# CONFIG_SOC_SERIES_KINETIS_KWX is not set
# CONFIG_SOC_SERIES_LPC54XXX is not set
# CONFIG_SOC_SERIES_LPC55XXX is not set
# CONFIG_SOC_SERIES_EFM32HG is not set
# CONFIG_SOC_SERIES_EFM32PG12B is not set
# CONFIG_SOC_SERIES_EFM32WG is not set
# CONFIG_SOC_SERIES_EFR32FG1P is not set
# CONFIG_SOC_SERIES_EFR32MG12P is not set
# CONFIG_SOC_SERIES_STM32F0X is not set
# CONFIG_SOC_SERIES_STM32F1X is not set
# CONFIG_SOC_SERIES_STM32F2X is not set
# CONFIG_SOC_SERIES_STM32F3X is not set
# CONFIG_SOC_SERIES_STM32F4X is not set
# CONFIG_SOC_SERIES_STM32F7X is not set
# CONFIG_SOC_SERIES_STM32G0X is not set
# CONFIG_SOC_SERIES_STM32H7X is not set
# CONFIG_SOC_SERIES_STM32L0X is not set
# CONFIG_SOC_SERIES_STM32L1X is not set
# CONFIG_SOC_SERIES_STM32L4X is not set
# CONFIG_SOC_SERIES_STM32MP1X is not set
# CONFIG_SOC_SERIES_STM32WBX is not set
# CONFIG_SOC_TI_LM3S6965 is not set
# CONFIG_SOC_SERIES_CC13X2_CC26X2 is not set
# CONFIG_SOC_SERIES_CC2650 is not set
# CONFIG_SOC_SERIES_CC32XX is not set
# CONFIG_SOC_SERIES_MSP432P4XX is not set

#
# Hardware Configuration
#
CONFIG_CPU_HAS_ARM_MPU=y
CONFIG_SOC_FAMILY="nordic_nrf"
CONFIG_SOC_FAMILY_NRF=y
CONFIG_HAS_HW_NRF_ACL=y
CONFIG_HAS_HW_NRF_CC310=y
CONFIG_HAS_HW_NRF_CCM=y
CONFIG_HAS_HW_NRF_CLOCK=y
CONFIG_HAS_HW_NRF_COMP=y
CONFIG_HAS_HW_NRF_ECB=y
CONFIG_HAS_HW_NRF_EGU0=y
CONFIG_HAS_HW_NRF_EGU1=y
CONFIG_HAS_HW_NRF_EGU2=y
CONFIG_HAS_HW_NRF_EGU3=y
CONFIG_HAS_HW_NRF_EGU4=y
CONFIG_HAS_HW_NRF_EGU5=y
CONFIG_HAS_HW_NRF_GPIO0=y
CONFIG_HAS_HW_NRF_GPIO1=y
CONFIG_HAS_HW_NRF_GPIOTE=y
CONFIG_HAS_HW_NRF_I2S=y
CONFIG_HAS_HW_NRF_LPCOMP=y
CONFIG_HAS_HW_NRF_MWU=y
CONFIG_HAS_HW_NRF_NFCT=y
CONFIG_HAS_HW_NRF_PDM=y
CONFIG_HAS_HW_NRF_POWER=y
CONFIG_HAS_HW_NRF_PPI=y
CONFIG_HAS_HW_NRF_PWM0=y
CONFIG_HAS_HW_NRF_PWM1=y
CONFIG_HAS_HW_NRF_PWM2=y
CONFIG_HAS_HW_NRF_PWM3=y
CONFIG_HAS_HW_NRF_QDEC=y
CONFIG_HAS_HW_NRF_QSPI=y
CONFIG_HAS_HW_NRF_RADIO_IEEE802154=y
CONFIG_HAS_HW_NRF_RNG=y
CONFIG_HAS_HW_NRF_RTC0=y
CONFIG_HAS_HW_NRF_RTC1=y
CONFIG_HAS_HW_NRF_RTC2=y
CONFIG_HAS_HW_NRF_SAADC=y
CONFIG_HAS_HW_NRF_SPI0=y
CONFIG_HAS_HW_NRF_SPI1=y
CONFIG_HAS_HW_NRF_SPI2=y
CONFIG_HAS_HW_NRF_SPIM0=y
CONFIG_HAS_HW_NRF_SPIM1=y
CONFIG_HAS_HW_NRF_SPIM2=y
CONFIG_HAS_HW_NRF_SPIM3=y
CONFIG_HAS_HW_NRF_SPIS0=y
CONFIG_HAS_HW_NRF_SPIS1=y
CONFIG_HAS_HW_NRF_SPIS2=y
CONFIG_HAS_HW_NRF_SWI0=y
CONFIG_HAS_HW_NRF_SWI1=y
CONFIG_HAS_HW_NRF_SWI2=y
CONFIG_HAS_HW_NRF_SWI3=y
CONFIG_HAS_HW_NRF_SWI4=y
CONFIG_HAS_HW_NRF_SWI5=y
CONFIG_HAS_HW_NRF_TEMP=y
CONFIG_HAS_HW_NRF_TIMER0=y
CONFIG_HAS_HW_NRF_TIMER1=y
CONFIG_HAS_HW_NRF_TIMER2=y
CONFIG_HAS_HW_NRF_TIMER3=y
CONFIG_HAS_HW_NRF_TIMER4=y
CONFIG_HAS_HW_NRF_TWI0=y
CONFIG_HAS_HW_NRF_TWI1=y
CONFIG_HAS_HW_NRF_TWIM0=y
CONFIG_HAS_HW_NRF_TWIM1=y
CONFIG_HAS_HW_NRF_TWIS0=y
CONFIG_HAS_HW_NRF_TWIS1=y
CONFIG_HAS_HW_NRF_UART0=y
CONFIG_HAS_HW_NRF_UARTE0=y
CONFIG_HAS_HW_NRF_UARTE1=y
CONFIG_HAS_HW_NRF_USBD=y
CONFIG_HAS_HW_NRF_WDT=y
CONFIG_SOC_NRF52840=y
# CONFIG_SOC_NRF52810_QFAA is not set
# CONFIG_SOC_NRF52811_QFAA is not set
# CONFIG_SOC_NRF52832_CIAA is not set
# CONFIG_SOC_NRF52832_QFAA is not set
# CONFIG_SOC_NRF52832_QFAB is not set
CONFIG_SOC_NRF52840_QIAA=y
CONFIG_SOC_DCDC_NRF52X=y
# CONFIG_NFCT_PINS_AS_GPIOS is not set
CONFIG_GPIO_AS_PINRESET=y
CONFIG_NRF_ENABLE_ICACHE=y
# end of Hardware Configuration

CONFIG_SOC_COMPATIBLE_NRF=y
CONFIG_SOC_COMPATIBLE_NRF52X=y

#
# ARM Options
#
CONFIG_CPU_CORTEX=y
CONFIG_CPU_CORTEX_M=y

#
# Architecture Floating Point Options
#
# end of Architecture Floating Point Options

CONFIG_CPU_CORTEX_M4=y
CONFIG_ISA_THUMB2=y
CONFIG_CPU_CORTEX_M_HAS_SYSTICK=y
CONFIG_CPU_CORTEX_M_HAS_BASEPRI=y
CONFIG_CPU_CORTEX_M_HAS_VTOR=y
CONFIG_CPU_CORTEX_M_HAS_PROGRAMMABLE_FAULT_PRIOS=y
CONFIG_ARMV7_M_ARMV8_M_MAINLINE=y
CONFIG_ARMV7_M_ARMV8_M_FP=y

#
# ARM Cortex-M options
#
CONFIG_LDREX_STREX_AVAILABLE=y
CONFIG_DATA_ENDIANNESS_LITTLE=y
CONFIG_STACK_ALIGN_DOUBLE_WORD=y
# CONFIG_RUNTIME_NMI is not set
CONFIG_FAULT_DUMP=2
CONFIG_XIP=y
# end of ARM Cortex-M options

#
# ARM Cortex-M0/M0+/M3/M4/M7/M23/M33 options
#
CONFIG_GEN_ISR_TABLES=y
# CONFIG_ZERO_LATENCY_IRQS is not set
CONFIG_PLATFORM_SPECIFIC_INIT=y
# end of ARM Cortex-M0/M0+/M3/M4/M7/M23/M33 options

CONFIG_ARM_MPU=y
CONFIG_ARM_MPU_REGION_MIN_ALIGN_AND_SIZE=32
# CONFIG_MPU_STACK_GUARD is not set
# CONFIG_MPU_ALLOW_FLASH_WRITE is not set
CONFIG_ARCH="arm"
# end of ARM Options

# CONFIG_ARC is not set
CONFIG_ARM=y
# CONFIG_X86 is not set
# CONFIG_X86_64 is not set
# CONFIG_NIOS2 is not set
# CONFIG_RISCV is not set
# CONFIG_XTENSA is not set
# CONFIG_ARCH_POSIX is not set

#
# General Architecture Options
#
# CONFIG_HW_STACK_PROTECTION is not set
# CONFIG_USERSPACE is not set
CONFIG_PRIVILEGED_STACK_SIZE=1024
CONFIG_PRIVILEGED_STACK_TEXT_AREA=256
CONFIG_KOBJECT_TEXT_AREA=256
# CONFIG_STACK_GROWS_UP is not set

#
# Interrupt Configuration
#
# CONFIG_DYNAMIC_INTERRUPTS is not set
CONFIG_GEN_IRQ_VECTOR_TABLE=y
CONFIG_GEN_SW_ISR_TABLE=y
CONFIG_ARCH_SW_ISR_TABLE_ALIGN=0
CONFIG_GEN_IRQ_START_VECTOR=0
# CONFIG_IRQ_OFFLOAD is not set
# end of Interrupt Configuration
# end of General Architecture Options

CONFIG_ARCH_HAS_STACK_PROTECTION=y
CONFIG_ARCH_HAS_USERSPACE=y
CONFIG_ARCH_HAS_EXECUTABLE_PAGE_BIT=y
CONFIG_ARCH_HAS_RAMFUNC_SUPPORT=y
CONFIG_ARCH_HAS_THREAD_ABORT=y
CONFIG_HAS_SYS_POWER_STATE_DEEP_SLEEP_1=y
CONFIG_CPU_HAS_FPU=y
CONFIG_CPU_HAS_MPU=y
CONFIG_MEMORY_PROTECTION=y
CONFIG_MPU_REQUIRES_POWER_OF_TWO_ALIGNMENT=y
# CONFIG_FLOAT is not set

#
# General Kernel Options
#
CONFIG_MULTITHREADING=y
CONFIG_NUM_COOP_PRIORITIES=16
CONFIG_NUM_PREEMPT_PRIORITIES=15
CONFIG_MAIN_THREAD_PRIORITY=0
CONFIG_COOP_ENABLED=y
CONFIG_PREEMPT_ENABLED=y
CONFIG_PRIORITY_CEILING=0
CONFIG_NUM_METAIRQ_PRIORITIES=0
# CONFIG_SCHED_DEADLINE is not set
# CONFIG_SCHED_CPU_MASK is not set
CONFIG_MAIN_STACK_SIZE=1024
CONFIG_IDLE_STACK_SIZE=320
CONFIG_THREAD_STACK_INFO=y
# CONFIG_THREAD_CUSTOM_DATA is not set
CONFIG_ERRNO=y
CONFIG_SCHED_DUMB=y
# CONFIG_SCHED_SCALABLE is not set
# CONFIG_SCHED_MULTIQ is not set
# CONFIG_WAITQ_SCALABLE is not set
CONFIG_WAITQ_DUMB=y

#
# Kernel Debugging and Metrics
#
# CONFIG_INIT_STACKS is not set
# CONFIG_KERNEL_DEBUG is not set
CONFIG_BOOT_BANNER=y
CONFIG_BOOT_DELAY=0
# CONFIG_EXECUTION_BENCHMARKING is not set
# CONFIG_THREAD_MONITOR is not set
# CONFIG_THREAD_NAME is not set
# end of Kernel Debugging and Metrics

#
# Work Queue Options
#
CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE=1024
CONFIG_SYSTEM_WORKQUEUE_PRIORITY=-1
CONFIG_OFFLOAD_WORKQUEUE_STACK_SIZE=1024
CONFIG_OFFLOAD_WORKQUEUE_PRIORITY=-1
# end of Work Queue Options

#
# Atomic Operations
#
CONFIG_ATOMIC_OPERATIONS_BUILTIN=y
# end of Atomic Operations

#
# Timer API Options
#
CONFIG_TIMESLICING=y
CONFIG_TIMESLICE_SIZE=0
CONFIG_TIMESLICE_PRIORITY=0
# CONFIG_POLL is not set
# end of Timer API Options

#
# Other Kernel Object Options
#
CONFIG_NUM_MBOX_ASYNC_MSGS=10
CONFIG_NUM_PIPE_ASYNC_MSGS=10
CONFIG_HEAP_MEM_POOL_SIZE=0
# end of Other Kernel Object Options

CONFIG_ARCH_HAS_CUSTOM_SWAP_TO_MAIN=y
CONFIG_SWAP_NONATOMIC=y
CONFIG_SYS_CLOCK_EXISTS=y

#
# Initialization Priorities
#
CONFIG_KERNEL_INIT_PRIORITY_OBJECTS=30
CONFIG_KERNEL_INIT_PRIORITY_DEFAULT=40
CONFIG_KERNEL_INIT_PRIORITY_DEVICE=50
CONFIG_APPLICATION_INIT_PRIORITY=90
# end of Initialization Priorities

#
# Security Options
#
# CONFIG_STACK_CANARIES is not set
# end of Security Options

#
# SMP Options
#
CONFIG_MP_NUM_CPUS=1
# CONFIG_SCHED_IPI_SUPPORTED is not set
# end of SMP Options

CONFIG_TICKLESS_IDLE=y
CONFIG_TICKLESS_IDLE_THRESH=3
CONFIG_TICKLESS_KERNEL=y
# CONFIG_SYS_POWER_DEEP_SLEEP_STATES is not set
# CONFIG_SYS_PM_STATE_LOCK is not set
# CONFIG_SYS_PM_DEBUG is not set
CONFIG_SYS_PM_POLICY_RESIDENCY=y
# CONFIG_SYS_PM_POLICY_DUMMY is not set
# CONFIG_SYS_PM_POLICY_APP is not set
CONFIG_SYS_PM_MIN_RESIDENCY_DEEP_SLEEP_1=60000
# CONFIG_DEVICE_POWER_MANAGEMENT is not set
# end of General Kernel Options

CONFIG_HAS_DTS=y
CONFIG_HAS_DTS_GPIO=y

#
# Device Drivers
#
# CONFIG_IEEE802154 is not set
CONFIG_UART_CONSOLE_ON_DEV_NAME="UART_0"
CONFIG_CONSOLE=y
CONFIG_CONSOLE_INPUT_MAX_LINE_LEN=128
CONFIG_CONSOLE_HAS_DRIVER=y
# CONFIG_CONSOLE_HANDLER is not set
CONFIG_UART_CONSOLE=y
CONFIG_UART_CONSOLE_INIT_PRIORITY=60
# CONFIG_UART_CONSOLE_DEBUG_SERVER_HOOKS is not set
# CONFIG_UART_CONSOLE_MCUMGR is not set
# CONFIG_USB_UART_CONSOLE is not set
# CONFIG_RAM_CONSOLE is not set
# CONFIG_IPM_CONSOLE_SENDER is not set
# CONFIG_IPM_CONSOLE_RECEIVER is not set
# CONFIG_UART_PIPE is not set
# CONFIG_UART_MCUMGR is not set
# CONFIG_SLIP is not set
# CONFIG_NET_LOOPBACK is not set
CONFIG_SERIAL=y

#
# Capabilities
#
CONFIG_SERIAL_HAS_DRIVER=y
CONFIG_SERIAL_SUPPORT_ASYNC=y
CONFIG_SERIAL_SUPPORT_INTERRUPT=y
# CONFIG_UART_ASYNC_API is not set
# CONFIG_UART_INTERRUPT_DRIVEN is not set
# CONFIG_UART_LINE_CTRL is not set
# CONFIG_UART_DRV_CMD is not set

#
# Serial Drivers
#
# CONFIG_UART_NSIM is not set
CONFIG_UART_NRFX=y
CONFIG_UART_0_NRF_UART=y
# CONFIG_UART_0_NRF_UARTE is not set
# CONFIG_UART_0_NRF_PARITY_BIT is not set
# CONFIG_UART_0_NRF_FLOW_CONTROL is not set
# CONFIG_UART_1_NRF_UARTE is not set
CONFIG_NRF_UART_PERIPHERAL=y
# CONFIG_UART_ALTERA_JTAG is not set

#
# Interrupt Controllers
#
# CONFIG_MULTI_LEVEL_INTERRUPTS is not set
# end of Interrupt Controllers

#
# Timer Drivers
#
# CONFIG_CORTEX_M_SYSTICK is not set
# CONFIG_SYSTEM_CLOCK_DISABLE is not set
# CONFIG_TIMER_READS_ITS_FREQUENCY_AT_RUNTIME is not set
# CONFIG_SYSTEM_CLOCK_SLOPPY_IDLE is not set
CONFIG_SYSTEM_CLOCK_INIT_PRIORITY=0
CONFIG_TICKLESS_CAPABLE=y
# end of Timer Drivers

# CONFIG_ENTROPY_GENERATOR is not set
# CONFIG_GPIO_SHELL is not set
# CONFIG_GPIO_DW is not set
# CONFIG_GPIO_SCH is not set
CONFIG_GPIO_NRFX=y
CONFIG_GPIO_NRF_INIT_PRIORITY=40
CONFIG_GPIO_NRF_P0=y
CONFIG_GPIO_NRF_P1=y
# CONFIG_GPIO_INTEL_APL is not set
# CONFIG_SHARED_IRQ is not set
# CONFIG_I2C is not set
# CONFIG_I2S is not set
# CONFIG_PWM is not set
# CONFIG_PINMUX is not set
# CONFIG_ADC is not set
# CONFIG_RTC is not set
CONFIG_CLOCK_CONTROL_NRF=y
# CONFIG_CLOCK_CONTROL_NRF_K32SRC_EXT_LOW_SWING is not set
# CONFIG_CLOCK_CONTROL_NRF_K32SRC_EXT_FULL_SWING is not set
# CONFIG_PTP_CLOCK is not set
# CONFIG_IPM is not set
# CONFIG_FLASH is not set
# CONFIG_COUNTER is not set
# CONFIG_DMA is not set
# CONFIG_USB is not set
# CONFIG_CRYPTO is not set
# CONFIG_DISPLAY is not set
# CONFIG_LED_STRIP is not set
# CONFIG_WIFI is not set
# CONFIG_LED is not set
# CONFIG_CAN is not set
# CONFIG_MODEM is not set
# CONFIG_AUDIO is not set
# CONFIG_NEURAL_NET_ACCEL is not set
# CONFIG_HWINFO is not set
# CONFIG_ESPI is not set
# end of Device Drivers

#
# C Library
#
# CONFIG_NEWLIB_LIBC is not set
CONFIG_STDOUT_CONSOLE=y
CONFIG_MINIMAL_LIBC_MALLOC_ARENA_SIZE=0
# CONFIG_MINIMAL_LIBC_LL_PRINTF is not set
# end of C Library

#
# Additional libraries
#
# CONFIG_LVGL is not set

#
# OS Support Library
#
# CONFIG_JSON_LIBRARY is not set
# CONFIG_RING_BUFFER is not set
# CONFIG_BASE64 is not set
# end of OS Support Library

CONFIG_POSIX_MAX_FDS=4
# CONFIG_POSIX_API is not set
# CONFIG_UPDATEHUB is not set
# end of Additional libraries

# CONFIG_BT is not set
# CONFIG_CONSOLE_SUBSYS is not set
# CONFIG_CPLUSPLUS is not set

#
# System Monitoring Options
#
# CONFIG_PERFORMANCE_METRICS is not set
# CONFIG_STATS is not set
# end of System Monitoring Options

#
# Debugging Options
#
# CONFIG_DEBUG is not set
# CONFIG_TRACING is not set
# CONFIG_STACK_USAGE is not set
# CONFIG_STACK_SENTINEL is not set
CONFIG_PRINTK=y
CONFIG_EARLY_CONSOLE=y
# CONFIG_ASSERT is not set
# CONFIG_FORCE_NO_ASSERT is not set
# CONFIG_OBJECT_TRACING is not set
# CONFIG_OVERRIDE_FRAME_POINTER_DEFAULT is not set
# CONFIG_DEBUG_INFO is not set
# CONFIG_OPENOCD_SUPPORT is not set
# CONFIG_TRACING_CPU_STATS is not set
# CONFIG_TRACING_CTF is not set
CONFIG_HAS_SEGGER_RTT=y
# CONFIG_USE_SEGGER_RTT is not set
# end of Debugging Options

# CONFIG_DISK_ACCESS is not set

#
# File Systems
#
# CONFIG_FILE_SYSTEM is not set
CONFIG_FS_LITTLEFS_NUM_FILES=4
CONFIG_FS_LITTLEFS_NUM_DIRS=4
CONFIG_FS_LITTLEFS_READ_SIZE=16
CONFIG_FS_LITTLEFS_PROG_SIZE=16
CONFIG_FS_LITTLEFS_CACHE_SIZE=64
CONFIG_FS_LITTLEFS_LOOKAHEAD_SIZE=32
CONFIG_FS_LITTLEFS_BLOCK_CYCLES=512
# CONFIG_NVS is not set
# end of File Systems

# CONFIG_LOG is not set

#
# Management
#
# CONFIG_MCUMGR_SMP_BT is not set
# CONFIG_MCUMGR_SMP_SHELL is not set
# CONFIG_MCUMGR_SMP_UART is not set
# CONFIG_MCUMGR is not set
# end of Management

#
# Networking
#
# CONFIG_NET_BUF is not set
# CONFIG_NETWORKING is not set
# end of Networking

# CONFIG_SHELL is not set
# CONFIG_IMG_MANAGER is not set
# CONFIG_TEST_RANDOM_GENERATOR is not set

#
# Storage
#
# end of Storage

# CONFIG_SETTINGS is not set

#
# Testing
#
# CONFIG_ZTEST is not set
# CONFIG_TEST is not set
# CONFIG_TEST_SHELL is not set
CONFIG_TEST_EXTRA_STACKSIZE=0
# CONFIG_TEST_USERSPACE is not set
# end of Testing

# CONFIG_CHARACTER_FRAMEBUFFER is not set
# CONFIG_JWT is not set

#
# External Sources
#

#
# HALs
#
CONFIG_HAS_CMSIS=y
# CONFIG_HAS_MEC_HAL is not set
CONFIG_HAS_NRFX=y
# end of HALs

#
# Cryptography
#
# CONFIG_TINYCRYPT is not set
# end of Cryptography

# CONFIG_FNMATCH is not set
# end of External Sources

#
# Build and Link Features
#

#
# Linker Options
#
CONFIG_KERNEL=y
# CONFIG_LINKER_ORPHAN_SECTION_PLACE is not set
CONFIG_LINKER_ORPHAN_SECTION_WARN=y
# CONFIG_LINKER_ORPHAN_SECTION_ERROR is not set
# CONFIG_CODE_DATA_RELOCATION is not set
CONFIG_HAS_FLASH_LOAD_OFFSET=y
# CONFIG_USE_CODE_PARTITION is not set
CONFIG_FLASH_LOAD_OFFSET=0
CONFIG_FLASH_LOAD_SIZE=0
# CONFIG_HAVE_CUSTOM_LINKER_SCRIPT is not set
# CONFIG_CUSTOM_RODATA_LD is not set
# CONFIG_CUSTOM_RWDATA_LD is not set
# CONFIG_CUSTOM_SECTIONS_LD is not set
# CONFIG_LINK_WHOLE_ARCHIVE is not set
CONFIG_KERNEL_ENTRY="__start"
CONFIG_LINKER_SORT_BY_ALIGNMENT=y
# end of Linker Options

#
# Compiler Options
#
# CONFIG_NATIVE_APPLICATION is not set
CONFIG_SIZE_OPTIMIZATIONS=y
# CONFIG_SPEED_OPTIMIZATIONS is not set
# CONFIG_DEBUG_OPTIMIZATIONS is not set
# CONFIG_NO_OPTIMIZATIONS is not set
CONFIG_COMPILER_OPT=""
# end of Compiler Options

#
# Build Options
#
CONFIG_KERNEL_BIN_NAME="zephyr"
CONFIG_OUTPUT_STAT=y
CONFIG_OUTPUT_DISASSEMBLY=y
CONFIG_OUTPUT_PRINT_MEMORY_USAGE=y
CONFIG_BUILD_OUTPUT_BIN=y
# CONFIG_BUILD_OUTPUT_EXE is not set
# CONFIG_BUILD_OUTPUT_S19 is not set
# CONFIG_BUILD_NO_GAP_FILL is not set
# CONFIG_BUILD_OUTPUT_STRIPPED is not set
# CONFIG_APPLICATION_DEFINED_SYSCALL is not set
# end of Build Options
# end of Build and Link Features

#
# Boot Options
#
# CONFIG_IS_BOOTLOADER is not set
# CONFIG_BOOTLOADER_MCUBOOT is not set
# CONFIG_REBOOT is not set
# CONFIG_MISRA_SANE is not set
# end of Boot Options

#
# Compatibility
#
CONFIG_COMPAT_INCLUDES=y
# end of Compatibility
`